docs/how-to/hip_runtime_api/call_stack.rst

```diff
@@ -0,0 +1,132 @@
+based on specific kernel requirements. This helps prevent stack overflow errors
+by ensuring sufficient stack memory is allocated.
+
+.. code-block:: cpp
+        
+    #include <hip/hip_runtime.h>
+    #include <iostream>
+
+    #define HIP_CHECK(expression)                \
+    {                                            \
+        const hipError_t status = expression;    \
+        if(status != hipSuccess){                \
+                std::cerr << "HIP error "        \
+                    << status << ": "            \
+                    << hipGetErrorString(status) \
+                    << " at " << __FILE__ << ":" \
+                    << __LINE__ << std::endl;    \
+        }                                        \
+    }
+
+    int main()
+    {
+        size_t stackSize;
+        HIP_CHECK(hipDeviceGetLimit(&stackSize, hipLimitStackSize));
+        std::cout << "Default stack size: " << stackSize << " bytes" << std::endl;
+
+        // Set a new stack size
+        size_t newStackSize = 1024 * 8; // 1MB
+        HIP_CHECK(hipDeviceSetLimit(hipLimitStackSize, newStackSize));
+        std::cout << "New stack size set to: " << newStackSize << " bytes" << std::endl;
+
+        HIP_CHECK(hipDeviceGetLimit(&stackSize, hipLimitStackSize));
+        std::cout << "Updated stack size: " << stackSize << " bytes" << std::endl;
+
+        return 0;
+    }
+
+Handling recursion and deep function calls
+--------------------------------------------------------------------------------
+
+Similar to CPU programming, recursive functions and deeply nested function calls
+are supported. However, developers must ensure that these functions do not
+exceed the available stack memory, considering the limited resources on GPUs.
+
+.. code-block:: cpp
+
+    #include <hip/hip_runtime.h>
+    #include <iostream>
+    
+    #define HIP_CHECK(expression)                \
+    {                                            \
+        const hipError_t status = expression;    \
+        if(status != hipSuccess){                \
+                std::cerr << "HIP error "        \
+                    << status << ": "            \
+                    << hipGetErrorString(status) \
+                    << " at " << __FILE__ << ":" \
+                    << __LINE__ << std::endl;    \
+        }                                        \
+    }
+    
+    __device__ unsigned long long factorial(unsigned long long n)
+    {
+        if (n == 0 || n == 1) {
+            return 1;
+        }
+        return n * factorial(n - 1);
+    }
+    
+    __global__ void kernel(unsigned long long n)
+    {
+        unsigned long long result = factorial(n);
+        const size_t x = threadIdx.x + blockDim.x * blockIdx.x;
+        
+        if( x == 0)
+          printf("%llu! = %llu \n", n, result);
+    }
+    
+    int main()
+    {
+        kernel<<<1, 1>>>(10);
+        HIP_CHECK(hipDeviceSynchronize());
```

Comment:
For some reason this didn't return an error for me, even when setting it to an unreasonable high number (the program crashed instead).

Also, while this is a nice example, it's not entirely suitable. 21! is already big enough to not fit in a 64 bit integer any more. Using doubles somewhat makes this better, but that way the result still reaches "infinity" before the stack overflow occurs.

Reply:
I'm thinking about it. Thanks!

Reply:
As a suggestion: https://en.wikipedia.org/wiki/1_%2B_2_%2B_3_%2B_4_%2B_%E2%8B%AF

Reply:
Good idea!
I added the Fibonacci sequence. What do you think, is it okay, or should it be triangular numbers?